runtime-modules/project-token/src/types.rs

```diff
@@ -1,32 +1,48 @@
+use sp_std::iter::Sum;
+
 use codec::{Decode, Encode, MaxEncodedLen};
-use common::{bloat_bond::RepayableBloatBond, MembershipTypes};
-use frame_support::storage::bounded_btree_map::BoundedBTreeMap;
+use common::{
+    bloat_bond::RepayableBloatBond, numerical::one_plus_interest_pow_fixed, MembershipTypes,
+};
 use frame_support::{
     dispatch::{fmt::Debug, DispatchError, DispatchResult},
     ensure,
-    traits::Get,
+    traits::{tokens::Balance as BalanceTrait, Get},
+    BoundedBTreeMap, BoundedVec,
 };
 use scale_info::TypeInfo;
 #[cfg(feature = "std")]
 use serde::{Deserialize, Serialize};
-use sp_arithmetic::traits::{AtLeast32BitUnsigned, One, Saturating, Unsigned, Zero};
+use sp_arithmetic::traits::{AtLeast32BitUnsigned, One, Saturating, Zero};
 use sp_runtime::{
-    traits::{CheckedAdd, CheckedMul, CheckedSub, Convert, Hash, UniqueSaturatedInto},
-    PerThing, Permill, Perquintill, SaturatedConversion,
+    traits::Hash, FixedPointNumber, FixedPointOperand, FixedU128, Permill, Perquintill,
 };
 use sp_std::{
     borrow::ToOwned,
     cmp::{max, min},
     collections::btree_map::BTreeMap,
     convert::{TryFrom, TryInto},
-    iter::Sum,
+    ops::Add,
     vec::Vec,
 };
 use storage::{BagId, DataObjectCreationParameters};
 
 // crate imports
 use crate::{errors::Error, Config, RepayableBloatBondOf};
 
+// trait "aliases"
+// `BlockNumber` will be implemented as `u64` in the runtime configuration
```

Comment:
The runtime implements BlockNumber as u32, not u64. Is this assumption here problematic?